third-party/realdds/src/dds-network-adapter-watcher.cpp

```diff
@@ -72,8 +72,14 @@ class network_adapter_watcher_singleton
     ~network_adapter_watcher_singleton()
     {
         _adapter_watcher.reset();  // signal the thread to finish
-        if( _th.joinable() )
-            _th.join();
+        try {
+            if (_th.joinable())
+                _th.join();
+        }
+        catch (std::system_error&)
+        {
+            std::cerr << "join failed, thread might no longer be joinable";
```

Comment:
Please use LOG_ERROR

Reply:
And, if there's an error, there's an error message too, no?

Reply:
done